api/db-cdr/generate-cdr/import-gcs-data.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+set -xeuo pipefail
+IFS=$'\n\t'
+# Imports data from gcs bucket into local mysql cdr<VERSION> tables
+# Example usage with ./project.rb :
+# ./project.rb generate-cloudsql-cdr --account peter.speltz@pmi-ops.org --cdr-version 20180201 --bucket all-of-us-workbench-cloudsql-create
+
+USAGE="./generate-cdr/import-gcs-data.sh --account <ACCOUNT> --bucket <BUCKET>  --cdr-db-name <cdrYYYYMMDD>"
+while [ $# -gt 0 ]; do
+  echo "1 is $1"
+  case "$1" in
+    --account) ACCOUNT=$2; shift 2;;
+    --bucket) BUCKET=$2; shift 2;;
+    --cdr-db-name) CDR_DB_NAME=$2; shift 2;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+# Todo this requires args in right order and doesn't print usage. Prints "Unbound variable ...."
+if [ -z "${ACCOUNT}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${BUCKET}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+if [ -z "${CDR_DB_NAME}" ]
+then
+  echo "Usage: $USAGE"
+  exit 1
+fi
+
+drop_index_activity="-PrunList=drop_bigdata_indexes"
+add_index_activity="-PrunList=add_bigdata_indexes"
+export CDR_DB_NAME  # export this so other scripts and gradle knows this
+echo "Importing data from $BUCKET"
+# Ruby is not installed in our dev container and this script is short, so bash is fine.
+
+# Download data and import
+REMOTE_DATA_LOC=https://storage.googleapis.com/$BUCKET
+echo "Importing data files from $REMOTE_DATA_LOC"
+
+# Add tables names of files to import here
+TABLES=(achilles_analysis achilles_results achilles_results_concept domain vocabulary criteria concept concept_relationship)
+
+# Make a dir for the csvs
+local_fpath=./generate-cdr/tmp/$CDR_DB_NAME
+rm -rf $local_fpath
+mkdir -p $local_fpath
+
+# Download data and truncate tables before dropping indexes
+gsutil -m cp gs://$BUCKET/$CDR_DB_NAME/*.csv $local_fpath
+
+# Drop indexes before importing
+echo "Dropping indexes on big tables to speed up inserts"
+../gradlew update $drop_index_activity
+
+# Import data
+for table in "${TABLES[@]}"
+do
+  # Truncate the table in case this is ever run separate
+  mysql -h ${DB_HOST} --port ${DB_PORT} -u root -p${MYSQL_ROOT_PASSWORD} -e "truncate table $CDR_DB_NAME.$table"
+
+  #  some files come in multiple csv because big. So we have to check for that
+  shopt -s nullglob # This makes files be an empty array if there aren't any matching files
```

Comment:
Is it expected we might have no files for certain tables?

Reply:
Not expected, however I use the fact that I get an empty array to check if i have more than one file for table or just one file named $table.csv

So i first check for $table0*.csv and if i get an array matching that , process those. Otherwise process $table.csv.  Maybe a better way to do it. Let me see ...

Ok. I think i made it better. If table.csv exists, table0*.csv wont exist. Logic now uses that fact.